grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -337,7 +337,15 @@ class CycleCountService {
                 'in'("id", ids)
             }
         } as List<CycleCount>
-        return cycleCounts.collect { CycleCountDto.toDto(it) }
+
+        return cycleCounts.collect { cycleCount ->
+            def dto = CycleCountDto.toDto(cycleCount)
+            dto.cycleCountItems = dto.cycleCountItems.sort { a, b ->
+                def comparison = a.inventoryItem?.expirationDate <=> b.inventoryItem?.expirationDate
+                comparison == 0 ? a.inventoryItem?.lotNumber <=> b.inventoryItem?.lotNumber : comparison
+            }
+            return dto
+        }
```

Comment:
as discussed on Slack, this should be done in the `createCriteria` closure. The way it's done now, is somewhat a "static" sorting, and we'd want the sorting on the query side.